storage/txcache/wrappedTransaction.go

```diff
@@ -3,17 +3,44 @@ package txcache
 import (
 	"bytes"
 
+	"github.com/ElrondNetwork/elrond-go/core/atomic"
 	"github.com/ElrondNetwork/elrond-go/data"
 )
 
 const estimatedSizeOfBoundedTxFields = uint64(128)
 
 // WrappedTransaction contains a transaction, its hash and extra information
 type WrappedTransaction struct {
-	Tx              data.TransactionHandler
-	TxHash          []byte
-	SenderShardID   uint32
-	ReceiverShardID uint32
+	Tx                     data.TransactionHandler
+	TxHash                 []byte
+	SenderShardID          uint32
+	ReceiverShardID        uint32
+	isImmuneToEvictionFlag atomic.Flag
+}
+
+// GetKey gets the transaction hash
+func (wrappedTx *WrappedTransaction) GetKey() []byte {
+	return wrappedTx.TxHash
+}
+
+// Payload gets the inner (wrapped) transaction
+func (wrappedTx *WrappedTransaction) Payload() interface{} {
```

Comment:
the payload of any inner transaction should implement transactionHandler interface. it could return that.

Reply:
I removed the `Payload()` from the `CacheItem` interface (part 3 of the fix-after-review).